homeassistant/components/sharkiq/vacuum.py

```diff
@@ -198,6 +217,35 @@ async def async_locate(self, **kwargs: Any) -> None:
         """Cause the device to generate a loud chirp."""
         await self.sharkiq.async_find_device()
 
+    async def async_clean_room(self, rooms: list[str], **kwargs: Any) -> None:
+        """Clean a specific room."""
+        if len(rooms) == 0:
+            raise exceptions.HomeAssistantError("No rooms to clean were provided.")
+        rooms_to_clean = []
+        valid_rooms = []
+        if self.available_rooms is not None:
+            valid_rooms = self.available_rooms
+        all_rooms_reachable = True
+        for room in rooms:
+            if room in valid_rooms:
+                rooms_to_clean.append(room)
+            elif room.capitalize() in valid_rooms:
+                rooms_to_clean.append(room.capitalize())
+            else:
+                all_rooms_reachable = False
+                LOGGER.error("Room not reachable: %s", room)
```

Comment:
```suggestion
                LOGGER.error("Invalid room %s", room)
                raise exceptions.HomeAssistantError(
                    "One or more of the rooms listed are unavailable to your vacuum. "
                    "Make sure all rooms match the Shark App, including capitalization."
                )
```

As you raise an error, the service will be flagged as failed.